engine/access/state_stream/backend/handler.go

```diff
@@ -199,6 +199,16 @@ func (h *Handler) SubscribeEvents(request *executiondata.SubscribeEventsRequest,
 	}
 }
 
-func (h *Handler) GetRegisterValues(ctx context.Context, request *executiondata.GetRegisterValuesRequest) (*executiondata.GetRegisterValuesResponse, error) {
-	return nil, status.Error(codes.Unimplemented, "not implemented")
+func (h *Handler) GetRegisterValues(_ context.Context, request *executiondata.GetRegisterValuesRequest) (*executiondata.GetRegisterValuesResponse, error) {
+	// Convert data
+	registerIDs, err := convert.MessagesToRegisterIDs(request.GetRegisterIds())
+	if err != nil {
+		return nil, status.Errorf(codes.InvalidArgument, "could not convert register IDs: %v", err)
+	}
+	// get payload from store
+	values, err := h.api.GetRegisterValues(registerIDs, request.GetBlockHeight())
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "could not get register values: %v", err)
```

Comment:
This will convert all errors to `Internal`. `ConvertError` only updates the code if one wasn't set
```suggestion
		return nil, rpc.ConvertError(err, "could not get register values", codes.Internal)
```